### PR TITLE
Add zintermediate utility

### PR DESCRIPTION
**Pull request by dadrian**
This adds an executable that takes in a large number (e.g. 120K) of
possible intermediates, and checks to see if they're actually
intermediates.

### CODE REVIEW

File: verifier/verifier.go [hunk 1]
```diff
@@ -104,6 +104,7 @@ func (v *Verifier) convertOptions(opt *VerificationOptions) (out x509.VerifyOpti
 	out.Roots = v.Roots
 	out.Intermediates = v.Intermediates
 	out.DNSName = opt.Name
+	out.KeyUsages = []x509.ExtKeyUsage{x509.ExtKeyUsageAny}
```
Comment (zakird) [inline verifier/verifier.go]: feels like this should be called ExtendedKeyUsages given that KeyUsages exists and is something else.

Comment (dadrian) [inline verifier/verifier.go]: This matches x509.VerifyOptions, and tbh, should probably be in a different commit.

File: x509/zintermediate/zintermediate.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	}
+	inputFileName = flag.Arg(0)
+
+	inputFormatArg = strings.ToLower(inputFormatArg)
+	switch inputFormatArg {
+	case "base64":
+		inputFormat = inputFormatBase64
+	case "pem":
+		inputFormat = inputFormatPEM
+	case "json":
```
Comment (zakird) [inline x509/zintermediate/zintermediate.go]: oh gawd why

Comment (dadrian) [inline x509/zintermediate/zintermediate.go]: Because you get certs from Censys. It would just read `raw`, if I got around to it.

File: verifier/verifier.go (hunk 1 above)
Comment (zakird) [inline verifier/verifier.go]: What calls this and why is this set here? It's not called by the other code?

File: x509/zintermediate/zintermediate.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	if err != nil {
+		log.Fatalf("could not open %s: %s", inputFileName, err)
+	}
+	rootPool, err := loadPEMPool(rootFile)
+	rootFile.Close()
+	if err != nil {
+		log.Fatalf("could not load roots: %s", err)
+	}
+
+	log.Infof("loading candidate intermediates from %s", inputFileName)
```
Comment (zakird) [inline x509/zintermediate/zintermediate.go]: This should be on line 122 given the location of L117.

File: verifier/verifier.go (hunk 1 above)
Comment (dadrian) [inline verifier/verifier.go]: Nothing calls this right now. I'm going to remove this change from this commit.
